src/BlockTypes/ProductImageGallery.php

```diff
@@ -28,39 +28,142 @@ protected function get_block_type_uses_context() {
+		 *
+		 * @since TBD
+		 */
+		$wrapper_classes = apply_filters(
+			'woocommerce_single_product_image_gallery_classes',
+			array(
+				'woocommerce-product-gallery',
+				'woocommerce-product-gallery--' . ( $post_thumbnail_id ? 'with-images' : 'without-images' ),
+				'woocommerce-product-gallery--columns-' . absint( $columns ),
+				'images',
+			)
+		);
+
+		if ( $post_thumbnail_id ) {
+			$product_image_gallery_placeholder = wc_get_gallery_image_html( $post_thumbnail_id, true );
+		} else {
+			$product_image_gallery_placeholder = sprintf(
+				'<div class="woocommerce-product-gallery__image--placeholder"><img src="%1$s" alt="%2$s" class="wp-post-image" /></div></div>',
+				esc_url( wc_placeholder_img_src( 'woocommerce_single' ) ),
+				esc_html__( 'Awaiting product image', 'woo-gutenberg-products-block' )
+			);
+		}
+		/**
+		 * The woocommerce_single_product_image_thumbnail_html filter.
+		 *
+		 * Allow users to modify the product gallery thumbnail html.
+		 *
+		 * @since TBD
+		 */
+		$product_image_gallery_placeholder = apply_filters( 'woocommerce_single_product_image_thumbnail_html', $product_image_gallery_placeholder, $post_thumbnail_id );
+
+		return sprintf(
+			'<div class="%1$s" data-columns="%2$s"><div class="woocommerce-product-gallery__wrapper">%3$s %4$s</div></div>',
+			esc_attr( implode( ' ', array_map( 'sanitize_html_class', $wrapper_classes ) ) ),
+			esc_attr( $columns ),
+			$product_image_gallery_placeholder,
+			$this->product_thumbnails( $product )
+		);
+	}
+
+	/**
+	 * Based on the single-product/product-thumbnails.php template from the core of WooCommerce.
+	 *
+	 * @param  \WC_Product $product Product instance.
+	 *
+	 * @return false|string
+	 */
+	private function product_thumbnails( $product ) {
+		$attachment_ids = $product->get_gallery_image_ids();
+		if ( ! $attachment_ids || ! $product->get_image_id() ) {
+			return '';
+		}
+
+		ob_start();
+		foreach ( $attachment_ids as $attachment_id ) {
+			echo wp_kses_post( wc_get_gallery_image_html( $attachment_id ) );
+		}
+		return ob_get_clean();
+	}
+
 	/**
 	 * Include and render the block.
 	 *
-	 * @param array    $attributes Block attributes. Default empty array.
-	 * @param string   $content    Block content. Default empty string.
-	 * @param WP_Block $block      Block instance.
+	 * @param array     $attributes Block attributes. Default empty array.
+	 * @param string    $content    Block content. Default empty string.
+	 * @param \WP_Block $block      Block instance.
+	 *
 	 * @return string Rendered block type output.
 	 */
 	protected function render( $attributes, $content, $block ) {
-
 		$post_id = $block->context['postId'];
-		global $product;
-		$product = wc_get_product( $post_id );
+
+		if ( ! isset( $post_id ) ) {
+			return '';
+		}
+
+		$single_post = get_post( $post_id );
+		if ( ! $single_post instanceof \WP_Post ) {
+			return '';
+		}
+
+		$single_product = wc_get_product( $post_id );
```

Comment:
The `single` prefix on the variable name is redundant. You can still use `product`.